### PR TITLE
Add EventHandler javadoc

### PR DESCRIPTION
**Pull request by Xephi**
There was no informations for priority() and ignoreCancelled() method

### CODE REVIEW

**Comment by riking** [comment]
This has spaces on the end of lines, can you fix it to match our style guide?

File: src/main/java/org/bukkit/event/EventHandler.java [hunk 1]
```diff
@@ -11,7 +11,21 @@
 @Target(ElementType.METHOD)
 @Retention(RetentionPolicy.RUNTIME)
 public @interface EventHandler {
+    
+    /**
+     * Define the priority of the event,
+     * 
+     * First execution to the last: LOWEST -> LOW -> NORMAL -> HIGH -> HIGHEST -> MONITOR
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/event/EventHandler.java]: Should be an ordered list and it also is longer than 78 characters.

File: src/main/java/org/bukkit/event/EventHandler.java [hunk 2]
```diff
@@ -11,7 +11,21 @@
...
+     * 
+     */
     EventPriority priority() default EventPriority.NORMAL;
-
+    
+    
+    /**
+     * Define if we must ignore if the event is cancelled or not
+     * 
+     * true -> ignoreCancelled / false -> Do not ignoreCancelled
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/event/EventHandler.java]: Considering that javadocs mirror HTML, this is illegible; it does not convey the intended meaning.

**Comment by Wolvereness** [comment]
On the note of HTML, the line breaks in the javadoc do not appear as you'd expect them to. If you intend for multi-paragraph situation, then the use of `<p \>` is recommended.

**Comment by Xephi** [comment]
I'll do these change , sorry for convenient x)

Xephi

File: src/main/java/org/bukkit/event/EventHandler.java [hunk 3]
```diff
@@ -11,7 +11,26 @@
...
+     * <p />
+     * First Priority to the last priority executed:
+     * <p />
+     * <ol>
+     * <li>LOWEST</li>
+     * <li>LOW</li>
+     * <li>NORMAL</li>
+     * <li>HIGH</li>
+     * <li>HIGHEST</li>
+     * </ol>
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/event/EventHandler.java:27]: Missing monitor.

File: src/main/java/org/bukkit/event/EventHandler.java [hunk 4]
```diff
@@ -11,7 +11,26 @@
...
+     */
     EventPriority priority() default EventPriority.NORMAL;
-
+    
+    /**
+     * Define if we must ignore if the event is cancelled or not.
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/event/EventHandler.java:32]: Ignore what? This javadoc isn't indicating what this actually does.

If you ignoreCancelled = true, that means your method will not get called for any cancelled event. ignoreCancelled = false implies your method will _always_ get called.

File: src/main/java/org/bukkit/event/EventHandler.java [hunk 5]
```diff
@@ -11,7 +11,27 @@
...
+     * <p />
+     * If ignoreCancelled = true , isCancelled() method is ignore.
+     */
     boolean ignoreCancelled() default false;
-}
+}
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/event/EventHandler.java]: Always newline at end of file

File: src/main/java/org/bukkit/event/EventHandler.java [hunk 6]
```diff
@@ -11,7 +11,27 @@
...
+     * <li>LOW</li>
+     * <li>NORMAL</li>
+     * <li>HIGH</li>
+     * <li>HIGHEST</li>
+     * <li>MONITOR</li>
+     * </ol>
+     */
     EventPriority priority() default EventPriority.NORMAL;
-
+
```
Comment (Wolvereness) [inline src/main/java/org/bukkit/event/EventHandler.java]: There is trailing whitespace here.

File: src/main/java/org/bukkit/event/EventHandler.java [hunk 7]
```diff
@@ -11,7 +11,30 @@
...
+     */
     EventPriority priority() default EventPriority.NORMAL;
 
+    /**
+     * Define if we must ignore if the event is cancelled or not.
+     * <p />
+     * If ignoreCancelled = true and the event is cancelled,
+     * your method is not call.
+     * <p />
+     * Otherwise, your method is always call.
```
Comment (SpaceManiac) [inline src/main/java/org/bukkit/event/EventHandler.java]: Wrong tense on "call"?

**Pull request merged**